tests/std/tests/VSO_0157762_feature_test_macros/test.cpp

```diff
@@ -5,465 +5,431 @@
 
 int main() {} // COMPILE-ONLY
 
-// COMPILER FEATURE-TEST MACROS
+// ATTRIBUTE FEATURE-TEST MACROS.
 
-#ifdef _MSVC_LANG
-#if _MSVC_LANG > 201703L
-#define CXX17_MODE 1
-#define CXX20_MODE 1
-#elif _MSVC_LANG > 201402L
-#define CXX17_MODE 1
-#define CXX20_MODE 0
+#ifdef __has_cpp_attribute
+// Good.
 #else
-#define CXX17_MODE 0
-#define CXX20_MODE 0
+#error Expected __has_cpp_attribute to be defined.

```

Comment:
In general, I think I'd prefer the more direct "`meow` is not `woof`" to "Expected `meow` to be `woof`" with the implied "...which it is not". (If you don't want to rewrite all of these messages a dozen times, I suggest giving other reviewers a chance to check in before making any sweeping changes.)

Reply:
Well there's also inverse cases so would you want "`meow` is `woof`" too? 
In the current form I capitalised NOT as I thought it could easily be missed. Then again with the shorter form it'll be harder to miss? 🤷‍♂️